solarforecastarbiter/reference_forecasts/forecast.py

```diff
@@ -0,0 +1,235 @@
+       ghi_clear: :py:func:`cloud_cover_to_ghi_linear`
+    4. Estimate cloudy sky DNI and DHI using the Erbs model.
+
+    Don't use this function if you already have solar position.
+
+    Parameters
+    ----------
+    latitude : float
+    longitude : float
+    elevation : float
+    cloud_cover : Series
+        Cloud cover in %.
+    solar_position_func : function
+
+    Returns
+    -------
+    ghi : pd.Series, dni : pd.Series, dhi : pd.Series
+
+    See also
+    --------
+    cloud_cover_to_irradiance_clearsky_scaling
+    """
+    solar_position = solar_position_func(
+        latitude, longitude, elevation, cloud_cover.index)
+    cs = pvmodel.calculate_clearsky(latitude, longitude, elevation,
+                                    solar_position['apparent_zenith'])
+    ghi, dni, dhi = cloud_cover_to_irradiance_clearsky_scaling(
+        cloud_cover, cs['ghi'], solar_position['zenith'])
+    return ghi, dni, dhi
+
+
+def cloud_cover_to_irradiance_clearsky_scaling_site(site, cloud_cover):
+    """
+    Estimates irradiance from cloud cover in the following steps:
+
+    1. Calculate solar position for the site.
+    2. Determine clear sky GHI using Ineichen model and climatological
+       turbidity.
+    3. Estimate cloudy sky GHI using a function of cloud_cover and
+       ghi_clear: :py:func:`cloud_cover_to_ghi_linear`
+    4. Estimate cloudy sky DNI and DHI using the Erbs model.
+
+    Don't use this function if you already have solar position.
+
+    Parameters
+    ----------
+    site : datamodel.Site
+    cloud_cover : Series
+        Cloud cover in %.
+
+    Returns
+    -------
+    ghi : pd.Series, dni : pd.Series, dhi : pd.Series
+
+    See also
+    --------
+    cloud_cover_to_irradiance_clearsky_scaling
+    """
+    solar_position = pvmodel.calculate_solar_position(
+        site.latitude, site.longitude, site.elevation, cloud_cover.index)
+    cs = pvmodel.calculate_clearsky(site.latitude, site.longitude,
+                                    site.elevation,
+                                    solar_position['apparent_zenith'])
+    ghi, dni, dhi = cloud_cover_to_irradiance_clearsky_scaling(
+        cloud_cover, cs['ghi'], solar_position['zenith'])
+    return ghi, dni, dhi
+
+
+def resample_args(*args, freq='1h'):
+    # this one uses map for fun
+    def f(arg):
+        if arg is None:
+            return None
+        else:
+            return arg.resample(freq).mean()
+    return list(map(f, args))
+
+
+def resample(arg, freq='1h'):
+    if arg is None:
+        return None
+    else:
+        return arg.resample(freq).mean()
+
+
+def interpolate_args(*args, freq='15min'):
+    # could add how kwarg to resample_args and lookup method with
+    # getattr but this seems much more clear
+    # this one uses a list comprehension for different fun
+    resampled_args = [
+        arg.resample(freq).interpolate()
+        for arg in args if arg is not None]
```

Comment:
While I prefer list comprehensions to map, this drops the None args

Reply:
hm I thought you'd like map for the future (ha) possibility of more delayed computations. 

The list comprehension could call the other function or could one line it like `[None if arg is None else arg.resample(freq).interpolate() for arg in args]`.